spec/helpers/activity_helper_spec.rb

```diff
@@ -62,6 +65,137 @@
     end
   end
 
+  describe "#link_to_add_comment" do
+    context "when activity is a programme" do
+      context "when user is authorised to add programme comments" do
+        it "returns button to add comment" do
+          authorise_creating_comments(commentable_type: :programme_activity, authorise: true)
```

Comment:
can we move these `authorise`s into the relevant `context` block? e.g. this one can go in `when user is authorised to add programme comments`

Reply:
I'm not sure that this is possible - I think it needs to be inside `it`. I get an error about it not being defined on the class if I move the method out of the `it` block and similar if I put the `allow` code directly there:

```ruby
undefined local variable or method `view' for #<Class:0x0000000114b0b7f0>
```

Reply:
Actually, I could probably just stick it in a `before` block where repeated